### ISSUE TITLE
Enhance Debugger : allow to (re)categorize methods from the debugger

### ISSUE THREAD

**Issue by demarey** [guillep/pharo-core#9918]
When using TDD mode, the debugger pops up on a MNU.
Then, you create the method and you can categorize it.
If you do a typo, you cannot see it and furthermore, you cannot modify the category from the debugger.
It would be nice to allow that.

**Comment by MarcusDenker** [comment]
I wonder: I find the forced menu very intrusive... I just do not want to think about which category to put the method at that point.
(and I do not have enough information: I need to be able to browse existing categories to take a good decision).

So after we add a way to change the category from the debugger, I think we could even remove that dialog and create the new method without a category at the start.

**Comment by demarey** [comment]
> I wonder: I find the forced menu very intrusive... I just do not want to think about which category to put the method at that point.
> (and I do not have enough information: I need to be able to browse existing categories to take a good decision).
> 
> So after we add a way to change the category from the debugger, I think we could even remove that dialog and create the new method without a category at the start.

I fully agree. Except for obvious categories (initialization, accessing) I need more context to choose. Maybe the debugger could set these 2 categories automatically (easy to guess) and do not categorize other methods but let the user a way to change the category and a quick way to browser the current method (A browse current method is missing)

**Comment by StevenCostiou** [comment]
@MarcusDenker There is an option in the settings for that (quick TDD I think) which if enabled removes all menus when defining method, and then no popup methods are autocategorized if possible or put in no category.

However I had to put that as a disabled option as many people wanted to have the category choice back.

**Comment by StevenCostiou** [comment]
@demarey It could be nice to have that, the hard part is to find where to display that information and how to change it.
Perhaps just above the code pane?

After that, I don't know, I have the feeling that categorizing methods is not really something to be done from the debugger even in TDD. 
To do that you need so much context, that in the end you need the class browser.

**Comment by StevenCostiou** [comment]
> @MarcusDenker There is an option in the settings for that (quick TDD I think) which if enabled removes all menus when defining method, and then no popup methods are autocategorized if possible or put in no category.
> 
> However I had to put that as a disabled option as many people wanted to have the category choice back.

To be precise: the option just auto-categorize (I think I don't remember) and in worst case => no category. 
It does not allow for choosing a category from the debugger.
This could easily be done from an extension pane but I don't know if the extension pane is going to be kept.

**Comment by Ducasse** [comment]
There is an automatic categorizer that is guessing for you but if we do not call it then it will not do it magically.

**Comment by StevenCostiou** [comment]
@Ducasse But it does with the quick TDD option in the debugger. 
Sometimes, the automatic categorizer just cannot find a category, but it works great for many cases.

The option is disabled by default because lots of people complained of not having the category choice popup...
I personnally always activate the quick TDD option because I want my methods to be created very fast without interference while doing TDD.

**Comment by StevenCostiou** [comment]
Sorry what I mean is: it uses the automatic categorizer, it is not the job of the debugger to choose categories ;)
Making this as a default is an interesting thing to discuss.

**Comment by demarey** [comment]
@StevenCostiou 

> @demarey It could be nice to have that, the hard part is to find where to display that information and how to change it.
> Perhaps just above the code pane?

In the stack table, we have method and package, why not add the category there?

> After that, I don't know, I have the feeling that categorizing methods is not really something to be done from the debugger even in TDD.
> To do that you need so much context, that in the end you need the class browser.

If there is basic auto-categorization + display of the category +  a quick access to browse the method, it would be enough I think.
The last point is missing. Selecting an entry in the stack, then  browse, then method is too slow. I would like a browse button just aside debugging buttons (step and co). WDYT?

**Comment by StevenCostiou** [comment]
PR done in NewTools => https://github.com/pharo-spec/NewTools/pull/308
These improvements could be back-ported without problems to P9.

**Comment by adri09070** [comment]
The PR was closed without any explanation. Is there any reason for that ?

It looks like it has been accidentally closed because the dev branch that was aimed has been deleted

**Comment by MarcusDenker** [comment]
Hmm.. that seems like a mistake, that close. I will re-open

**Comment by MarcusDenker** [comment]
Hmm... the PR seems to not exist anymore to re-open.
